konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/api/protocol/URIResolver.java

```diff
@@ -0,0 +1,49 @@
+package ai.konduit.serving.pipeline.api.protocol;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.*;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+
+public class URIResolver {
+
+    public static boolean isUrl(String input) {
+        if (input.startsWith("http://") ||
+            input.startsWith("https://") ||
+            input.startsWith("ftp://")) {
+            return true;
+        }
+        return false;
+    }
+
+    private static File cacheDirectory;
+
+    public static File getCacheDirectory() {
+        if (cacheDirectory == null) {
+            File f = new File(System.getProperty("user.home"), ".konduit_cache/");
```

Comment:
Let's also make this configurable (this is a good default though). Doing that in the next PR is fine.